consensus/ethash/sealer.go

```diff
@@ -46,18 +47,53 @@ var (
 	errInvalidSealResult = errors.New("invalid or stale proof-of-work solution")
 )
 
+// makeFakeDelay uses the ethash.threads value as a mean time (lambda)
+// for a Poisson distribution, returning a random value from
+// that discrete function. I think a Poisson distribution probably
+// fairly accurately models real world block times.
+// Note that this is a hacky way to use ethash.threads since
+// lower values will yield faster blocks, but it saves having
+// to add or modify any more code than necessary.
+func (ethash *Ethash) makeFakeDelay() float64 {
+	p := distuv.Poisson{
```

Comment:
Also we may want to rethink the "fake delay" language. IIRC there is a "fake delayer" for testing which basically adds a `time.Sleep` for block verification (but without actually running verification). Disambiguation here may be good.

Reply:
hmm, this simulates something similar to the difficulty. So will it make sense to name it **similar** to `makeFakeDifficultySimulation`?

Reply:
So I think we can use the word "Fake" to mean that `mix` and `nonce` don't need to be legit (are skipped during header verification). The "delay" part is the tricky part.

- https://github.com/etclabscore/core-geth/blob/master/consensus/ethash/ethash.go#L622. Here, "delay" means delay on the verification side. This is only used once, in a test, delaying for one `time.Millisecond`.
- The "delay" 180 proposes is at the _emission_ process, and is intended to model real-world emission pattern (ie Poisson).  Maybe `Ethash#ModePoissonFake`?

Reply:
I am working on adding a new `Mode: Ethash#ModePoissonFake`. The only question so far, is how will this mode gets enabled @meowsbits?

1. not clean, when `--dev --dev.[ethash|pow] --fakepow` instead of `Ethash#ModeFake` always use `Ethash#ModePoissonFake`
2. introduce a new cli flag `--fakepoissonpow`
3. change `--fakepow` to string arg accepting following modes [fake|poisson|full]

Reply:
A `Mode Ethash#ModePoissonFake` has been added at #180.
The enablement of this mode has to be decided, and it will probably happen in this PR.

Reply:
Yea. Tricky one. We're into what seems like sub-flag territory. Dev mode, with pick-your-poison consensus style; and fakepow mode, with pick-your-poisson emission style. Both have existing defaults. 

Here's what I suggest, for the sake of keeping the CLI API fairly stable, and using consistent namespacing.

- `--dev` behavior is unchanged. Uses Clique.
- `--dev.pow`. A new flag. A new mode. Incompatible with `--dev`.
- `--fakepow`. Behavior is added so that this flag actually does something when `geth` runs. (_ie_ the `SetEthash` addition).
- `--fakepow.poisson`. A new flag. A new fake-consensus type. Incompatible with `--fakepow`, but available and useable in the same places and ways.

This leaves the decision What ethash mode to use as a default for `--dev.pow`? I am suggesting to use not-fake PoW as the default, but, as @BelfordZ suggested, toggle the equivalent of `--mine --miner.threads=1`. The `--fakepow[.poisson]` flag(s) would be _applicable_, just not be enabled by default. 

This keeps the mode as "simple" as possible while also keeping options open for the user.

Reply:
@meowsbits all the above have been added. The only consideration is that `--dev.pow --fakepow` doesn't sealing as I reverted your changes at #180 to the original code from go-ethereum and added your changes only for the new `ModePoissonFake`. Check [here](https://github.com/etclabscore/core-geth/blob/c3b5cb551ab95de3832a5e5e6808296458fc0fb1/consensus/ethash/sealer.go#L70-L106).

I think I misunderstood your intentions on #180. Do you want me to bring back your change so as sealing works for `--fakepow`?

Reply:
> `--dev.pow --fakepow` doesn't sealing

Does this mean that it won't automatically start mining blocks? If so, then yes, I'd suggest we change that it does start automatically mining. Or, I'm not sure what you mean by sealing doesn't work for `fakepow`...?

Reply:
Updating here based on our discussion.

`--fakepow` is used only for the verify process and disables sealing of blocks (actually it create dump blocks), this should not be changed as it will affect the actual use case of `--fakepow`. For this reason `--dev.pow --fakepow` will not be supported as implemented https://github.com/etclabscore/core-geth/pull/243/commits/6eb9ce9e7b0ef097880deb3b65b616959a549077.

The supported modes will be:

1. `--dev.pow`: starting an Ethash Ephemeral chain
2. `--dev.pow --fakepow.poisson`: starting an Ethash Ephemeral chain, without verification of blocks and with mining delay (Poisson) based on --miner.threads